internal/db/schema/migrations/oss/postgres/88/01_worker_auth_invariant_trigger.up.sql

```diff
@@ -0,0 +1,42 @@
+-- Copyright (c) HashiCorp, Inc.
+-- SPDX-License-Identifier: BUSL-1.1
+
+begin;
+-- update_worker_auth_authorized is a before update trigger function for the
+-- worker_auth_authorized table. The worker_auth_authorized table is a child
+-- table of server_worker. A row contains a set encryption keys for a
+-- server_worker that are unique to that worker. A server_worker can only have
+-- two rows in the worker_auth_authorized table: one with a state of 'current'
+-- and one with the state of 'previous'.
+--
+--- This trigger function ensures that there is only ever one entry for
+--  with a state of 'current' and one with a state of 'previous'.
+create function update_worker_auth_authorized() returns trigger
+as $$
+begin
+    if new.state = 'current' then
+        perform
+        from worker_auth_authorized
+          where state = 'current';
```

Comment:
Isn't this per-worker? Should there be a `where` clause in here somewhere? Does this work if there are multiple workers?

Reply:
my bad, I've been too focused thinking about a single instance 😮‍💨  thanks!